shared/common-adapters/channel-container.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+import {Channel, type OwnProps} from './channel'
+import {connect, type TypedState} from '../util/container'
+import * as ChatGen from '../actions/chat-gen'
+import {pathSelector} from '../actions/route-tree'
+
+const mapStateToProps = (state: TypedState) => {
+  return {
+    currentPath: pathSelector(state),
+  }
+}
+
+const mapDispatchToProps = (dispatch, {convID}: OwnProps) => ({
+  onClick: currentPath =>
+    dispatch(
+      ChatGen.createSelectOrPreviewConversation({conversationIDKey: convID, previousPath: currentPath})
+    ),
+})
+
+const mergeProps = ({currentPath}, {onClick}, ownProps: OwnProps) => {
+  return {
+    ...ownProps,
+    onClick: () => onClick(currentPath),
+  }
+}
+
+// $FlowIssue
```

Comment:
Flow doesn't seem to "see" the `key` property being passed into the connected `Channel`.